aggregator/src/aggregation/rlc/gates.rs

```diff
@@ -0,0 +1,248 @@
+            || Value::known(Fr::one()),
+        )?;
+        b.copy_advice(|| "c", region, self.phase_2_column, *offset + 2)?;
+        let d = region.assign_advice(
+            || "d",
+            self.phase_2_column,
+            *offset + 3,
+            || a.value() + b.value(),
+        )?;
+        *offset += 4;
+
+        Ok(d)
+    }
+
+    /// Enforce res = a - b
+    pub(crate) fn sub(
+        &self,
+        region: &mut Region<Fr>,
+        a: &AssignedCell<Fr, Fr>,
+        b: &AssignedCell<Fr, Fr>,
+        offset: &mut usize,
+    ) -> Result<AssignedCell<Fr, Fr>, Error> {
+        self.selector.enable(region, *offset)?;
+
+        let res = region.assign_advice(
+            || "a",
+            self.phase_2_column,
+            *offset,
+            || a.value() - b.value(),
+        )?;
+        region.assign_advice(
+            || "b",
+            self.phase_2_column,
+            *offset + 1,
+            || Value::known(Fr::one()),
+        )?;
+        b.copy_advice(|| "c", region, self.phase_2_column, *offset + 2)?;
+        a.copy_advice(|| "d", region, self.phase_2_column, *offset + 3)?;
+        *offset += 4;
+
+        Ok(res)
+    }
+
+    /// Enforce res = a * b
+    pub(crate) fn mul(
+        &self,
+        region: &mut Region<Fr>,
+        a: &AssignedCell<Fr, Fr>,
+        b: &AssignedCell<Fr, Fr>,
+        offset: &mut usize,
+    ) -> Result<AssignedCell<Fr, Fr>, Error> {
+        self.selector.enable(region, *offset)?;
+
+        a.copy_advice(|| "a", region, self.phase_2_column, *offset)?;
+        b.copy_advice(|| "b", region, self.phase_2_column, *offset + 1)?;
+        region.assign_advice(
+            || "c",
+            self.phase_2_column,
+            *offset + 2,
+            || Value::known(Fr::zero()),
+        )?;
+        let d = region.assign_advice(
+            || "d",
+            self.phase_2_column,
+            *offset + 3,
+            || a.value() * b.value(),
+        )?;
+        *offset += 4;
+
+        Ok(d)
+    }
+
+    /// Enforce res = a * b + c
+    pub(crate) fn mul_add(
+        &self,
+        region: &mut Region<Fr>,
+        a: &AssignedCell<Fr, Fr>,
+        b: &AssignedCell<Fr, Fr>,
+        c: &AssignedCell<Fr, Fr>,
+        offset: &mut usize,
+    ) -> Result<AssignedCell<Fr, Fr>, Error> {
+        self.selector.enable(region, *offset)?;
+
+        a.copy_advice(|| "a", region, self.phase_2_column, *offset)?;
+        b.copy_advice(|| "b", region, self.phase_2_column, *offset + 1)?;
+        c.copy_advice(|| "c", region, self.phase_2_column, *offset + 2)?;
+        let d = region.assign_advice(
+            || "d",
+            self.phase_2_column,
+            *offset + 3,
+            || a.value() * b.value() + c.value(),
+        )?;
+        *offset += 4;
+
+        Ok(d)
+    }
+
+    /// caller need to ensure a is binary
+    /// return !a
+    pub(crate) fn not(
+        &self,
+        region: &mut Region<Fr>,
+        a: &AssignedCell<Fr, Fr>,
```

Comment:
It will be great if we can have a dedicated type `Boolean` for `a`. 
```
struct Boolean (AssignedCell<F, F>)
impl Boolean {
      fn new(cfg: &RlcConfig, region: &mut Region<F>, a: &AssignedCell<F, F>, offset: &mut usize) -> Self {
             let zero = cfg.mul_sub(region, a, a, a, offset);
             cfg.enforce_zero(region, zero);

             Self (a)
      }
}
```